File: npm/examples/basic.js
/**
 * Basic Example - Palindrome Checker API
 *
 * This example demonstrates how to use the Palindrome Checker API.
 * Make sure to set your API key in the .env file or replace '[YOUR_API_KEY]' below.
 */

require('dotenv').config();
const palindromecheckerAPI = require('../index.js');

// Initialize the API client
const api = new palindromecheckerAPI({
    api_key: process.env.API_KEY || '[YOUR_API_KEY]'
});

// Example query
var query = {
  text: "A man a plan a canal Panama",
  ignorecase: true,
  ignorespaces: true,
  ignorepunctuation: true
};

// Make the API request using callback
console.log('Making request to Palindrome Checker API...\n');

api.execute(query, function (error, data) {
    if (error) {
        console.error('Error occurred:');
        if (error.error) {
            console.error('Message:', error.error);
            console.error('Status:', error.status);
        } else {
            console.error(JSON.stringify(error, null, 2));
        }
        return;
    }

    console.log('Response:');
    console.log(JSON.stringify(data, null, 2));
});

File: examples/python/basic.py
"""
Palindrome Checker API - Basic Usage Example

This example demonstrates the basic usage of the Palindrome Checker API.
API Documentation: https://docs.apiverve.com/ref/palindromechecker
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/palindromechecker'

def call_palindromechecker_api():
    """
    Make a GET request to the Palindrome Checker API
    """
    try:
        # Query parameters
        params &#x3D; {&#x27;text&#x27;: &#x27;A man a plan a canal Panama&#x27;, &#x27;ignorecase&#x27;: true, &#x27;ignorespaces&#x27;: true, &#x27;ignorepunctuation&#x27;: true}

        headers = {
            'x-api-key': API_KEY
        }

        response = requests.get(API_URL, headers=headers, params=params)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Palindrome Checker API...\n')

    result = call_palindromechecker_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')
